keylime/tpm/tpm_main.py

```diff
@@ -1576,6 +1585,49 @@ def __add_boot_aggregate(log: dict) -> None:
                 except Exception:
                     pass
 
+    @staticmethod
+    def __unescape_eventlog(log: dict):
+        """
+        Newer versions of tpm2-tools escapes the YAML output and including the trailing null byte.
+        See: https://github.com/tpm2-software/tpm2-tools/commit/c78d258b2588aee535fd17594ad2f5e808056373
+        This converts it back to an unescaped string.
+        Example:
+            '"MokList\\0"' -> 'MokList'
+        """
+        if tpm.tools_version in ["3.2", "4.0", "4.2"]:
+            return
+
+        escaped_chars = [
+            ("\0", "\\0"),
+            ("\a", "\\a"),
+            ("\b", "\\b"),
+            ("\t", "\\t"),
+            ("\v", "\\v"),
+            ("\f", "\\f"),
+            ("\r", "\\r"),
+            ("\x1b", "\\e"),
+            ("'", "\\'"),
+            ("\\", "\\\\"),
+        ]
+
+        def recursive_unescape(data):
+            if isinstance(data, str):
+                if data.startswith('"') and data.endswith('"'):
+                    data = data[1:-1]
+                    for orig, escaped in escaped_chars:
+                        data = data.replace(escaped, orig)
+                    return data.rstrip("\0")
```

Comment:
the return can be removed, and you can `data = data.rstrip("\0")`

Reply:
yes fixed.